File: android/app/src/main/java/com/htahmasebi/phototransfer/protocol/ProtocolModels.kt
package com.htahmasebi.phototransfer.protocol

import kotlinx.serialization.Serializable

const val PROTOCOL_VERSION = 1

@Serializable
data class TransferManifest(
    val protocolVersion: Int,
    val files: List<ManifestFile>,
)

@Serializable
data class ManifestFile(
    val id: String,
    val name: String,
    val mediaType: String,
    val size: Long?,
)

@Serializable
data class InfoResponse(
    val protocolVersion: Int,
    val receiverName: String,
)

@Serializable
data class TransferCreatedResponse(
    val transferId: String,
)

@Serializable
data class CompleteResponse(
    val receivedFiles: Int,
)

File: mac/Sources/PhotoReceiver/Server/ProtocolModels.swift
import Foundation

enum TransferProtocol {
    static let version = 1
}

struct TransferManifest: Codable, Equatable, Sendable {
    let protocolVersion: Int
    let files: [ManifestFile]
}

struct ManifestFile: Codable, Equatable, Sendable {
    let id: String
    let name: String
    let mediaType: String
    let size: Int64?
}

struct InfoResponse: Codable, Equatable, Sendable {
    let protocolVersion: Int
    let receiverName: String
}

struct TransferCreatedResponse: Codable, Equatable, Sendable {
    let transferId: String
}

struct CompleteResponse: Codable, Equatable, Sendable {
    let receivedFiles: Int
}

struct TransferStatusResponse: Codable, Equatable, Sendable {
    let transferId: String
    let state: String
    let receivedFiles: Int
    let totalFiles: Int
}
